contract_manager/src/entities.ts

```diff
@@ -27,13 +35,64 @@ import { Contract, Storable } from "./base";
 
 export const Contracts: Record<string, Contract> = {};
 
+class InvalidTransactionError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "InvalidTransactionError";
+  }
+}
+
 export class SubmittedWormholeMessage {
   constructor(
     public emitter: PublicKey,
     public sequenceNumber: number,
     public cluster: string
   ) {}
 
+  static async fromTrasactionSignature(
```

Comment:
typo: fromTransactionSignature

also this could use a doc comment because it's not at all clear what this does. 

and I'm pretty sure the return type is actually Promise<SubmittedWormholeMessage> so no need for any?